install_files/ansible-base/roles/upgrade/files/0.3pre_upgrade.py

```diff
@@ -26,6 +32,24 @@ def backup_mon():
         t.add('/var/lib/tor/services/')
     print "**  Backed up system to {} before migrating.".format(tar_fn)
 
+def secure_unlink(path):
+    subprocess.check_call(['srm', '-r', path])
+
+def clean_large_deleted():
+    for source_dir in os.listdir(store_dir):
+        try:
+            source_id, journalist_designation = c.execute(
+                "SELECT id, journalist_designation FROM sources WHERE filesystem_id=?",
+                (source_dir,)).fetchone()
+            print source_id, journalist_designation, source_dir
+        except (sqlite3.Error, TypeError) as e:
```

Comment:
It would be better if this check were more explicit (general Python principle, "explicit is better than implicit"). We noticed the error because:

1. The `source_dir` was still present, because the `secure_unlink` had failed
2. The corresponding table in the DB had been deleted
3. As a result, the `SELECT` statement `WHERE filesystem_id=source_dir` returned `None`
4. A `TypeError` is raised when the line tries to unpack `None` into `source_id, journalist_designation`

Rather than catch the TypeError (which could theoretically be caused by something else), we should test explicitly to see if the given source is in the database. All exceptions should log errors and terminate execution (so we can inspect to see what is causing the issue and try to fix it up before the migration makes potentially destructive changes).